### PR TITLE
✨ feat: migrate less to token for Tag

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
https://github.com/ant-design/ant-design/issues/41884

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     migrate less to token for Tag     |
| 🇨🇳 Chinese |   迁移tag组件 less token     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 0c91a88</samp>

Refactor the tag component to use component tokens instead of less variables, and update the theme API and the migration guide accordingly. This improves the consistency and flexibility of the tag styles and the theming system.

### <samp>🤖 Generated by Copilot at 0c91a88</samp>

* Refactor tag component to use component tokens and expose them to theme API ([link](https://github.com/ant-design/ant-design/pull/42053/files?diff=unified&w=0#diff-cb2262b77ae285e7da02504971f378bec871797a530abbf2a34f14fdb3d92f85L152-R199), [link](https://github.com/ant-design/ant-design/pull/42053/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739R39-R49))
  - Add a third argument to `genComponentStyleHook` function that returns tag token values ([link](https://github.com/ant-design/ant-design/pull/42053/files?diff=unified&w=0#diff-cb2262b77ae285e7da02504971f378bec871797a530abbf2a34f14fdb3d92f85L152-R199))
  - Update `components/tag/style/index.ts` to use the new function and export the tag style hook ([link](https://github.com/ant-design/ant-design/pull/42053/files?diff=unified&w=0#diff-cb2262b77ae285e7da02504971f378bec871797a530abbf2a34f14fdb3d92f85L152-R199))
  - Add a new section to the migration guide for tag component in `docs/react/migrate-less-variables.zh-CN.md` ([link](https://github.com/ant-design/ant-design/pull/42053/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739R39-R49))
  - List the mapping between less variables and component tokens for tag styles ([link](https://github.com/ant-design/ant-design/pull/42053/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739R39-R49))

### CODE REVIEW

File: docs/react/migrate-less-variables.zh-CN.md [hunk 1]
```diff
@@ -36,3 +36,14 @@ title: less 变量迁移 Component Token
 | `@modal-confirm-body-padding` | - | 由于样式变化已废弃 |
 | `@modal-confirm-title-font-size` | `modalHeaderTitleFontSize` | - |
 | `@modal-border-radius` | `borderRadiusLG` | - |
+
+## Tag 标签
+
+<!-- prettier-ignore -->
+| less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@tag-border-radius` | - | 由于样式变化已废弃 |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:45]: borderRadiusSM

Comment (heiyu4585) [reply docs/react/migrate-less-variables.zh-CN.md]: done

File: components/tag/style/index.ts [hunk 2]
```diff
@@ -5,13 +5,15 @@ import { resetComponent } from '../../style';
...
 
-export interface ComponentToken {}
-
-interface TagToken extends FullToken<'Tag'> {
+export interface ComponentToken {
   tagFontSize: number;
   tagLineHeight: React.CSSProperties['lineHeight'];
   tagDefaultBg: string;
   tagDefaultColor: string;
+  tagBorderRadius: number;
```
Comment (MadCcc) [inline components/tag/style/index.ts:13]: 这个可以不用加，直接在表格中替换为对应使用的 token

File: docs/react/migrate-less-variables.zh-CN.md [hunk 3]
```diff
@@ -150,6 +150,17 @@ title: Less 变量迁移 Design Token
 
 <!-- ### Tag 标签 -->
 
+## Tag 标签
+
+<!-- prettier-ignore -->
+| less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@tag-border-radius` | `tagBorderlessBg` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:158]:
```suggestion
| `@tag-border-radius` | `borderRadiusSM` | - |
```

File: docs/react/migrate-less-variables.en-US.md [hunk 4]
```diff
@@ -150,6 +150,17 @@ This document contains the correspondence between all the less variables related
 
 <!-- ### Tag -->
 
+## Tag
+
+<!-- prettier-ignore -->
+| less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@tag-border-radius` | `tagBorderlessBg` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:158]:
```suggestion
| `@tag-border-radius` | `borderRadiusSM` | - |
```

File: docs/react/migrate-less-variables.en-US.md [hunk 5]
```diff
@@ -150,6 +150,17 @@ This document contains the correspondence between all the less variables related
 
 <!-- ### Tag -->
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:151]:
```suggestion
```

File: components/tag/style/index.ts (hunk 2 above)
Comment (heiyu4585) [reply components/tag/style/index.ts]: done

File: docs/react/migrate-less-variables.zh-CN.md (hunk 3 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.zh-CN.md]: done

File: docs/react/migrate-less-variables.en-US.md (hunk 4 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.en-US.md]: done

File: docs/react/migrate-less-variables.en-US.md (hunk 5 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.en-US.md:195]: done

File: docs/react/migrate-less-variables.zh-CN.md [hunk 6]
```diff
@@ -181,7 +181,16 @@ title: Less 变量迁移 Design Token
 
 <!-- ### Tabs 标签页 -->
 
-<!-- ### Tag 标签 -->
+## Tag 标签
+
+<!-- prettier-ignore -->
+| less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@tag-border-radius` | `borderRadiusSM` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:189]:
```suggestion
| `@tag-border-radius` | `borderRadiusSM` | Global Token |
```

File: docs/react/migrate-less-variables.zh-CN.md [hunk 7]
```diff
@@ -181,7 +181,16 @@ title: Less 变量迁移 Design Token
...
+| `@tag-default-bg` | `defaultBg` | - |
+| `@tag-default-color` | `defaultColor` | - |
+| `@tag-font-size` | `tagFontSize` | - |
+| `@tag-line-height` | `tagLineHeight` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:193]:
```suggestion
| `@tag-font-size` | `fontSizeSM` | Global Token |
| `@tag-line-height` | `lineHeightSM` | Global Token |
```

File: components/tag/style/index.ts [hunk 8]
```diff
@@ -149,31 +150,34 @@ const genBaseStyle = (token: TagToken): CSSInterpolation => {
 };
 
 // ============================== Export ==============================
-export default genComponentStyleHook('Tag', (token) => {
-  const { fontSize, lineHeight, lineWidth, fontSizeIcon } = token;
-  const tagHeight = Math.round(fontSize * lineHeight);
-
-  const tagFontSize = token.fontSizeSM;
-  const tagLineHeight = tagHeight - lineWidth * 2;
```
Comment (MadCcc) [inline components/tag/style/index.ts:157]: 这个直接用 lineHeightSM 应该就可以

File: docs/react/migrate-less-variables.zh-CN.md [hunk 9]
```diff
@@ -150,6 +150,17 @@ title: Less 变量迁移 Design Token
 
 <!-- ### Tag 标签 -->
```
[the remark this answers is not in the document]
Comment (heiyu4585) [reply docs/react/migrate-less-variables.zh-CN.md]: done

File: docs/react/migrate-less-variables.zh-CN.md (hunk 6 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.zh-CN.md]: done

File: components/tag/demo/component-token.tsx [hunk 10]
```diff
@@ -0,0 +1,37 @@
+import { CloseCircleOutlined, SyncOutlined } from '@ant-design/icons';
+import { ConfigProvider, Space, Tag } from 'antd';
+import React from 'react';
+
+export default () => (
+  <ConfigProvider
+    theme={{
+      components: {
+        Tag: {
+          tagFontSize: 20,
```
Comment (MadCcc) [inline components/tag/demo/component-token.tsx:10]: 这里要去掉咯

**Comment by MadCcc** [comment]
image
有点问题

File: components/tag/style/index.ts [hunk 11]
```diff
@@ -67,10 +65,10 @@ const genBaseStyle = (token: TagToken): CSSInterpolation => {
       height: 'auto',
       marginInlineEnd: token.marginXS,
       paddingInline,
-      fontSize: token.tagFontSize,
-      lineHeight: `${token.tagLineHeight}px`,
+      fontSize: token.fontSizeSM,
+      lineHeight: `${token.lineHeight}px`,
```
Comment (MadCcc) [inline components/tag/style/index.ts:69]:
```suggestion
      lineHeight: token.lineHeightSM,
```

File: components/tag/style/index.ts [hunk 12]
```diff
@@ -1,19 +1,17 @@
...
+  defaultColor: string;
+}
 
 interface TagToken extends FullToken<'Tag'> {
-  tagFontSize: number;
-  tagLineHeight: React.CSSProperties['lineHeight'];
-  tagDefaultBg: string;
-  tagDefaultColor: string;
-  tagIconSize: number;
-  tagPaddingHorizontal: number;
```
Comment (MadCcc) [inline components/tag/style/index.ts:16]: argos 还是有很多 diff，这里没有迁移成 component token 的都可以还原下

File: components/tag/style/index.ts [hunk 13]
```diff
@@ -67,10 +65,10 @@ const genBaseStyle = (token: TagToken): CSSInterpolation => {
       height: 'auto',
       marginInlineEnd: token.marginXS,
       paddingInline,
-      fontSize: token.tagFontSize,
-      lineHeight: `${token.tagLineHeight}px`,
+      fontSize: token.fontSizeSM,
+      lineHeight: token.lineHeight,
```
Comment (MadCcc) [inline components/tag/style/index.ts:69]:
```suggestion
      lineHeight: token.lineHeightSM,
```
主要是这里不对哈，建议还是把原来的自定义 token 加回去，看起来比较清楚些

Comment (MadCcc) [reply components/tag/style/index.ts:69]: 另外自定义 token 的 tag 前缀要留着

Comment (heiyu4585) [reply components/tag/style/index.ts]: done

File: components/tag/style/index.ts [hunk 14]
```diff
@@ -149,31 +149,31 @@ const genBaseStyle = (token: TagToken): CSSInterpolation => {
...
+      genTagStatusStyle(tagToken, 'warning', 'Warning'),
+    ];
+  },
+  (token) => {
+    const tagHeight = Math.round(token.fontSize * token.lineHeight);
+    const tagLineHeight = tagHeight - token.lineWidth * 2;
+    return {
+      defaultBg: token.colorFillQuaternary,
+      defaultColor: token.colorText,
+      tagLineHeight,
```
Comment (MadCcc) [inline components/tag/style/index.ts:176]: 是其他的还原回去哈，组件 token 不用加这个。另外这个取值应该就是 lineHeightSM

**Review by MadCcc — APPROVED** [review]

**Pull request merged**